File: 251001/Karpekov/lab5/TaskRedis/discussion/src/main/java/discussion/models/Post.java
package discussion.models;

import discussion.enums.PostState;
import org.springframework.data.annotation.Transient;
import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKey;
import org.springframework.data.cassandra.core.mapping.*;


@Table("tbl_post")
public class Post {
    @PrimaryKey
    private PostKey id;

    @Column
    private String content;

    public Post() {
    }

    public Post(PostKey id, String content) {
        this.id = id;
        this.content = content;
    }

    public PostKey getId() {
        return id;
    }

    public void setId(PostKey id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: 251001/Smychok/LAB3/CassandraPosts/Models/Post.cs
// Models/Post.cs
using Cassandra.Mapping.Attributes;

namespace CassandraPosts.Models;

[Table("tbl_post")]
public class Post
{
    [Column("post_id")]
    public Guid Id { get; set; }

    [Column("issue_id")]
    public int IssueId { get; set; }

    [Column("content")]
    public string Content { get; set; }

    [PartitionKey]
    [Column("country")]
    public string Country { get; set; } = "default"; // Для совместимости с тестами
}